src/backend/web/validation.js

```diff
@@ -49,5 +50,25 @@ function validateNewFeed() {
     }
   };
 }
+const postQueryValidationRules = [
+  query('per_page', 'per_page needs to be empty or a integer').custom(
+    (value) => !value || Number.isInteger(+value)
+  ),
+  query('page', 'page needs to be empty or an integer').custom(
+    (value) => !value || Number.isInteger(+value)
+  ),
+];
+const postQueryValidation = () => {
+  return async (req, res, next) => {
+    await Promise.all(postQueryValidationRules.map((rule) => rule.run(req)));
 
-module.exports.validateNewFeed = validateNewFeed;
+    const result = validationResult(req);
+    if (result.isEmpty()) {
+      return next();
+    }
+
+    const errors = result.array();
+    return res.status(400).send(errors);
+  };
+};
```

Comment:
Overall, this is good. You need to write some unit tests to test it.

Reply:
Do you not think the current unit test all test cases

Reply:
@zg3d : Only a couple of cases. Like the page query did not get tested. You need to do one for when the user provided both queries, one when they provided only of them. You can write at least 2 more tests in my opinions.

Reply:
the current tests test only the default value and per_page query